pkg/eval/pwd_test.go

```diff
@@ -0,0 +1,40 @@
+package eval
+
+import (
+	"errors"
+	"path/filepath"
+	"testing"
+)
+
+func TestBuiltinPwd(t *testing.T) {
+	tmpHome, cleanup := InTempHome()
+	defer cleanup()
+
+	mustMkdirAll("dir1")
+	mustMkdirAll("dir2")
+	dir1 := filepath.Join(tmpHome, "dir1")
+	dir2 := filepath.Join(tmpHome, "dir2")
+
+	Test(t,
+		That(`pwd=dir1 put $pwd; put $pwd`).Puts(dir1, tmpHome),
+		That(`cd $E:HOME/dir2; pwd=~ put $pwd; put $pwd`).Puts(tmpHome, dir2),
+		That(`cd $E:HOME/dir2; pwd=~/dir1 put $pwd; put $pwd`).Puts(dir1, dir2),
+		That(`cd $E:HOME/dir1; pwd=../dir2 put $pwd; put $pwd`).Puts(dir2, dir1),
```

Comment:
I don't see a good reason to alternate between E:HOME and ~ here; stick to the more idiomatic ~ if possible.

Reply:
Fair enough. Looking at it now I don't know why I did this other than due to the use of `InTempHome()`.